modules/idle-scheduler/src/lib.rs

```diff
@@ -77,13 +87,51 @@ pub mod module {
 	#[pallet::getter(fn next_task_id)]
 	pub type NextTaskId<T: Config> = StorageValue<_, Nonce, ValueQuery>;
 
+	///
+	#[pallet::storage]
+	#[pallet::getter(fn previous_relay_block)]
+	pub type PreviousRelayBlockNumber<T: Config> = StorageValue<_, BlockNumber, ValueQuery>;
+
 	#[pallet::pallet]
 	pub struct Pallet<T>(_);
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_initialize(_n: T::BlockNumber) -> Weight {
+			// Unwrap shouldn't fail if it does defaults to 0
+			let test: BlockNumber = T::RelayChainBlockNumberProvider::current_block_number()
+				.try_into()
+				.unwrap_or_default();
+			PreviousRelayBlockNumber::<T>::put(test);
+			0
+		}
+
 		fn on_idle(_n: T::BlockNumber, remaining_weight: Weight) -> Weight {
-			Self::do_dispatch_tasks(remaining_weight)
+			// Checks if we have skipped enough relay blocks without block production to skip dispatching
+			// scheduled tasks
+			let current_relay_block_number: BlockNumber = T::RelayChainBlockNumberProvider::current_block_number()
+				.try_into()
+				.unwrap_or_default();
+			let previous_relay_block_number = Self::previous_relay_block();
```

Comment:
```suggestion
			let previous_relay_block_number = PreviousRelayBlockNumber::<T>::take();
```

and no need to `kill`